### PR TITLE
Use codecov github action

### PR DESCRIPTION
**Pull request by ryanio**
Previously `codecov` didn't allow us to submit coverage reports using gh actions without secrets, which didn't allow us to report code coverage on PRs from forks, so we decided to use `circleci` just for coverage reporting.

Now that the [codecov github action](https://github.com/codecov/codecov-action#usage) allows for public repos to submit without a token we can remove our usage of `circleci`.

### CODE REVIEW

File: .github/workflows/account-build.yml [hunk 1]
```diff
@@ -33,8 +33,8 @@ jobs:
       - run: npx lerna run build --scope ethereumjs-account --include-dependencies
       - run: npx lerna run lint --scope ethereumjs-account
       - run: npm test
-        env:
-          CI: true
```
Comment (ryanio) [inline .github/workflows/account-build.yml:37]: I don't think these env `CI: true` were ever being used so I removed them.

**Comment by evertonfraga** [comment]
I am getting increasingly excited with our CI structure, now that all pieces seem to be coming together. We were waiting for this Codecov feature for months :)

With this new feature, we can safely drop CircleCI test runs, and integrate it on our GH actions. There are some other improvements we can tackle in this PR that Ryan and I have been discussed privately.

After integrating the codecov into our current GH actions, the combo of `paths-ignore` with `codecov/project` (total coverage) check might not work properly, as coverage tests would only report coverage for packages changed (and their dependents).

**A)** So one suggestion is to remove `codecov/project` check for PRs, while keeping it for builds on `master` only. The `codecov/patch` check is the most relevant for PRs anyway. There were some associated problems with `codecov/project` in the past (https://github.com/ethereumjs/ethereumjs-vm/pull/720#issuecomment-614913926). 

This solution will still leverage the `paths-ignore` rule, so not all packages' tests would run all the time.

**B)** If we choose to keep `codecov/project` check for PRs, we need to either remove the `paths-ignore`, or at the very minimum, run all code coverage tasks in a separate workflow, just like we did with CircleCI.
B.1) If we put all coverage tests in separate jobs, we'd still have the same 6 extra checks
B.2) If all coverage tests are run in sequence (to have a single check at the end), this job will certainly take long and it'll probably be counter-productive.

File: packages/account/package.json [hunk 2]
```diff
@@ -23,7 +22,7 @@
     "tsc": "ethereumjs-config-tsc",
     "lint": "ethereumjs-config-lint",
     "lint:fix": "ethereumjs-config-lint-fix",
-    "test": "npx tape -r ts-node/register test/*.spec.ts"
+    "test": "nyc tape -r ts-node/register test/*.spec.ts"
```
Comment (holgerd77) [inline packages/account/package.json:25]: I wonder if I would rather prefer to keep a separate `coverage` command here (or is this difficult due to changed conditions on the CI setup?). I am using `npm run test` locally all the time for test runs and I can imagine that I would find it rather distracting to have coverage reported all the time along. So for my taste this is binding things unnecessarily together.

**Comment by holgerd77** [comment]
@evertonfraga on first thought I would say A) is more practical here

File: packages/account/package.json (hunk 2 above)
Comment (evertonfraga) [reply packages/account/package.json:25]: Agreed!

**Comment by evertonfraga** [comment]
Remaining tasks for this PR:
-[ ] revert add `coverage` npm script in each of the 6 packages
-[ ] remove `nyc` command from test execution
-[ ] remove `codecov/project` check for PRs (keep `codecov/patch`)

**Comment by ryanio** [comment]
@evertonfraga great thanks I can do the first two, I don’t think I have the ability to remove the codecov/project pr check.

**Comment by evertonfraga** [comment]
By adding the `npm run coverage` on the test suite, you are running the same tests twice on the same run. `ethereumjs-config`'s  coverage script executes `npm run test`  underneath.

So, I'd advise to remove the `npm run test` steps on the jobs that have it.

— I also can do that later the day

File: .github/workflows/vm-build.yml [hunk 3]
```diff
@@ -22,51 +22,44 @@ name: VM
...
       - run: npx lerna bootstrap --scope ethereumjs-vm --ignore-scripts --include-dependencies --no-ci
-      - run: npx lerna run --scope ethereumjs-vm --include-dependencies build
-      - run: 'npm run test:blockchain'
-        env:
-          CI: true
-        working-directory: '${{ env.cwd }}'
-    env:
-      cwd: '${{github.workspace}}/packages/vm'
+      - run: npx lerna run build --scope ethereumjs-vm --include-dependencies
+      - run: npx lerna run test:blockchain --scope ethereumjs-vm
```
Comment (evertonfraga) [inline .github/workflows/vm-build.yml:65]: VM has a special test suite, but all good here 👍.

File: .github/workflows/vm-nightly-test.yml [hunk 4]
```diff
@@ -1,69 +1,54 @@
...
       - run: npx lerna bootstrap --scope ethereumjs-vm --ignore-scripts --include-dependencies --no-ci
-      - run: npx lerna run --scope ethereumjs-vm --include-dependencies build
-      - run: 'npm run test:state:slow'
-        env:
-          CI: true
-        working-directory: '${{ env.cwd }}'
-    env:
-      cwd: '${{github.workspace}}/packages/vm'
+      - run: npx lerna run build --scope ethereumjs-vm --include-dependencies
+      - run: npx lerna run test:state:slow --scope ethereumjs-vm
```
Comment (evertonfraga) [inline .github/workflows/vm-nightly-test.yml:54]: Only cleanup, no coverage needed.

File: packages/account/package.json [hunk 5]
```diff
@@ -23,7 +23,7 @@
     "tsc": "ethereumjs-config-tsc",
     "lint": "ethereumjs-config-lint",
     "lint:fix": "ethereumjs-config-lint-fix",
-    "test": "npx tape -r ts-node/register test/*.spec.ts"
+    "test": "tape -r ts-node/register test/*.spec.ts"
   },
   "keywords": [
     "ethereum",
```
Comment (evertonfraga) [inline packages/account/package.json:29]: OK

File: packages/block/package.json [hunk 6]
```diff
@@ -20,7 +20,7 @@
     "lint": "ethereumjs-config-lint",
     "lint:fix": "ethereumjs-config-lint-fix",
     "test": "npm run test:node && npm run test:browser",
-    "test:node": "npx tape -r ts-node/register test/*.spec.ts",
+    "test:node": "tape -r ts-node/register test/*.spec.ts",
     "test:browser:build": "tsc && cp test/testdata/*.json test-build/test/testdata",
     "test:browser": "npm run test:browser:build && karma start karma.conf.js"
   },
```
Comment (evertonfraga) [inline packages/block/package.json:26]: OK

File: packages/blockchain/package.json [hunk 7]
```diff
@@ -59,7 +58,6 @@
     "@types/node": "^11.11.4",
     "@types/semaphore": "^1.1.0",
     "@types/tape": "^4.2.33",
-    "coveralls": "^3.0.2",
     "nyc": "^13.0.1",
     "prettier": "^1.16.4",
     "tape": "^4.9.1",
```
Comment (evertonfraga) [inline packages/blockchain/package.json:63]: Always good to remove a dep 😎

File: packages/tx/package.json [hunk 8]
```diff
@@ -20,7 +19,7 @@
     "tsc": "ethereumjs-config-tsc",
     "lint": "ethereumjs-config-lint",
     "lint:fix": "ethereumjs-config-lint-fix",
-    "test": "npm run test:node",
+    "test": "npm run test:node && npm run test:browser",
```
Comment (evertonfraga) [inline packages/tx/package.json:22]: OK. I might want to undo this in the future for some other optimizations, but that's super OK for now.

**Comment by holgerd77** [comment]
Congrats on the merge here, cool! 😄  👍

**Pull request merged**